src/constraint_framework/4C_constraint_framework_submodelevaluator_nullspace.cpp

```diff
@@ -136,11 +150,12 @@ void Constraints::SubmodelEvaluator::NullspaceConstraintManager::evaluate_coupli
     }
   }
 
-  Q_dL_ = std::make_shared<Core::LinAlg::SparseMatrix>(*dof_map, constraint_space.num_vectors());
-
+  auto constraint_matrix = std::make_shared<Core::LinAlg::SparseMatrix>(
+      *dof_condition_map_, constraint_space.num_vectors());
   Core::LinAlg::multi_vector_to_linalg_sparse_matrix(
-      constraint_space, *dof_map, *constraint_map_, *Q_dL_);
+      constraint_space, *dof_condition_map_, *constraint_map_, *constraint_matrix);
```

Comment:
The constraint matrix is not a sparse matrix, rather a tall-and-skinny dense matrix, right? Why convert it to a sparse matrix?